lib/index.js

```diff
@@ -182,14 +181,18 @@ Mixpanel.prototype.track = function(track) {
   props = dates(props, iso);
   window.mixpanel.track(track.event(), props);
 
-  // register super properties for enhancedTrack
-  if (enhancedTrack && typeof superProps === 'object') {
-    window.mixpanel.register(superProps);
+  // register super properties if present in context.mixpanel.superProperties
+  if (superProps) {
+    var superPayload = {};
+    superProps.forEach(function(prop){superPayload[prop] = props[prop]});
+    window.mixpanel.register(superPayload);
   }
 
-  // set people properties for enhancedTrack
-  if (enhancedTrack && typeof peopleProps === 'object') {
-    window.mixpanel.people.set(peopleProps);
+  // set people properties if present in context.mixpanel.peopleProperties
+  if (peopleProps) {
+    var peoplePayload = {};
+    peopleProps.forEach(function(prop){peoplePayload[prop] = props[prop]});
```

Comment:
can't use `Array#forEach` unfortunately due to browser compat issues. let's try nathan's [pick](https://github.com/ndhoule/pick) component. something like:

```
var options = track.options(this.name);
var properties =  track.properties();

var superProps = pick((options.superProperties || []), properties);
var peopleProps = pick((options.peopleProperties || []), properties);

if (superProps.length) window.mixpanel.register(superProps);
if (peopleProps.length) window.mixpanel.register(peopleProps);
```